infrastructure/smart-contracts/contracts/drops/TinyKeysAirdrop.sol

```diff
@@ -133,4 +136,64 @@ contract TinyKeysAirdrop is Initializable, AccessControlUpgradeable {
             emit AirdropEnded();
         }
     }
+
+    
+    function processAirdropSegmentOnlyMint(uint256 _qtyToProcess) external onlyRole(DEFAULT_ADMIN_ROLE)  {
+        require(airdropStarted, "Airdrop not started");
+        require(airdropCounter <= totalSupplyAtStart, "Airdrop complete");
+        // Start where we left off
+        uint256 startingKeyId = airdropCounter;
+        // Ensure we don't go over the total supply
+        uint256 endingKeyId = Math.min(airdropCounter + _qtyToProcess, totalSupplyAtStart);     
+        // Connect to the referee and node license contracts
+        NodeLicense10 nodeLicense = NodeLicense10(nodeLicenseAddress);
+        // Loop through the range of node licenses
+        for (uint256 i = startingKeyId; i < endingKeyId; i++) {
+            // Determine the owner of each specific node license
+            address owner = nodeLicense.ownerOf(i);
+
+            uint256 start = nodeLicense.totalSupply() + 1;
+            // Mint the airdropped keys for the owner
+            nodeLicense.mintForAirdrop(keyMultiplier, owner);
```

Comment:
Instead of connecting to the node license 3 different times here. 

Should we update the mintForAirdrop function to accept keyId instead of owner? Since NodeLicense can look up the owner itself?

Reply:
Yes thats a very good idea, this will save some gas although I hope on mainnet everything will be much cheaper